### DISCUSSION TITLE
Error with sasioiom.py

### DISCUSSION THREAD

**Discussion by mrmorgan17** [sassoftware/saspy#525] [Q&A] [answered]
I was trying to connect to my SAS On Demand of Academics Session with the following code:

```
import saspy
sas = saspy.SASsession()
stat = sas.sasstat()
```

was running `stat.glm()` and `sas.submitLST()`

Specific `stat.glm()` code
```
sasfit = stat.glm(
  data = 'snapbean',
  cls = 'replicate',
  model = 'replicate S V S*V',
  manova = 'H = replicate / PRINTE PRINTH MSTAT=exact'
)
```

Python version 3.10.9 and saspy version 4.6.0

The procedure ran but I got the following:
```
/usr/local/lib/python3.10/site-packages/saspy/sasioiom.py:1009: UserWarning: Noticed 'ERROR:' in LOG, you ought to take a look and see if there was a problem
  lstf = lstf.decode()
```

And `sasfit` only had `ERROR_LOG` as an object

It seems that something is wrong with me `sasioiom.py` file. Any suggestions as to what to do?

**Comment by tomweber-sas** [comment]
Well, what was in the LOG, it says there's an error. What's the error? You can see either in the ERROR_LOG that was returned, or via print(sas.lastlog()) (right after running that) or print(sas.saslog()) for the whole sas log.

**Comment by mrmorgan17** [comment]
```
20                                                         The SAS System                      Monday, February  6, 2023 04:17:00 PM

663        
664        %macro proccall(d);
665        proc glm data=WORK.'snapbean'n  plots=all  ;
666        class replicate;
667        model replicate S V S*V;
668        manova H = replicate / PRINTE PRINTH MSTAT=exact;
669        run; quit; %mend;
670        %mangobj(glm0001,glm,'snapbean'n);
670                  proc glm data=WORK.'snapbean'n  plots=all  ; class replicate; model replicate S V S*V; manova H = replicate /
                                                                                                        _
                                                                                                        22
                                                                                                        76
670      ! PRINTE PRINTH MSTAT=exact; run; quit;
NOTE: The previous statement has been deleted.
ERROR: Effects used in the MANOVA statement must have appeared previously in the MODEL statement.
NOTE: The previous statement has been deleted.
ERROR 22-322: Syntax error, expecting one of the following: a name, -, :, =, _ALL_, _CHARACTER_, _CHAR_, _NUMERIC_.  

ERROR 76-322: Syntax error, statement will be ignored.

671        
672        
673        

21                                                         The SAS System                      Monday, February  6, 2023 04:17:00 PM
```

Here was my error log. It seems to not be liking the `*` character. I know I've used the `*` to denote interactions before and had no problem with that. Did syntax change?

**Comment by tomweber-sas** [comment]
Well, I wouldn't believe that the syntax has changed for a SAS proc; that's not something that we allow; no breaking changes. So, I'm assuming it's just a valid syntax error and that there's some other statement that's required to get rid of this error?
I have no idea whatsoever; I'm not a statistician and never use any procedures. I see the doc for this statement here, but clearly nothing about it tells me anything about what it's expecting: https://go.documentation.sas.com/doc/en/pgmsascdc/9.4_3.5/statug/statug_glm_syntax16.htm

Googling the error, I found this communities thread which, although not the same, may give you an idea of what this proc might be expecting or require to be able to run right: https://communities.sas.com/t5/SAS-Programming/Error-in-estimate-and-contrast-statements-need-help-interpreting/td-p/97200

The error seems to imply that you need to have previously references the variables being used, but I have no idea how to use the proc, so I can't provide the exact answer. At least, the code generated appears to be correct for what you coded up. So I expect it's just another thing you need to provide to get it to run.

**Comment by mrmorgan17** [comment, marked as answer]
It was just a syntax error on my end

**Reply by tomweber-sas** [reply]
ok, cool
